recipes-extend/sd-creator-linux/files/sd_flash.sh

```diff
@@ -0,0 +1,145 @@
+regex_y_check="\\b(Y|y)\\b"
+
+usr=$(id -u)
+echo "usr=$usr"
+if [ $usr != 0 ]; then
+	echo "Please run this script as sudo / root."
+	exit 0
+fi
+echo $regex_y_check
+if [ "${parted_present}" == "${dpkg_present}" ];
+then
+	echo "selected device is ${sd_dev}"
+	printf "Warning: all data will be erased on this drive. Please confirm (Y|y) : "
+	read -r format_confirm
+	if [[ ${format_confirm} =~ ${regex_y_check} ]]
+	then
+		echo "You have chosen to erase the memory device (${sd_dev})"
+		num_part=$(cat /proc/partitions | grep ${sd_dev} | wc -l)
+		echo $num_part
+		echo "Disk info:-------------------------------------------------------------------"
+		parted -s ${sd_dev} print
+		echo "-----------------------------------------------------------------------------"
+		echo "Unmounting device.."
+		df -hT | grep ${sd_dev} | cut -d " " -f 1 | xargs umount
+		if [ -z $(df -hT | grep ${sd_dev}) ]; then
+			echo "Unmount successful"
+		else
+			echo "Unmount failed."
+			exit
+		fi
+		echo "Erasing disk..."
+		#mounts=$(cat /proc/mounts | grep sdb | cut -d ' ' -f 2)
+		#cat ${mounts} | xargs umount
+		#echo "mounts = ${mounts}"
+		#echo "partiton 1 mount:"
+		sfdisk --delete ${sd_dev}
+		echo "Disk erase complete."
+		echo "-----------------------------------------------------------------------------"
+
+		echo "Disk info:"
+		parted -s ${sd_dev} print
+		echo "-----------------------------------------------------------------------------"
+		parted -s ${sd_dev} mklabel gpt
+		parted -s ${sd_dev} mkpart primary fat32 2MiB 512MiB
+		parted -s ${sd_dev} mkpart primary ext4 512MiB 4096MiB
+		parted -s ${sd_dev} align-check min 1
+		sync
+
+		echo "Partitioning complete"
+		echo "-----------------------------------------------------------------------------"
+		echo "Disk info:"
+		parted -s ${sd_dev} print
+		partprobe
+		echo "-----------------------------------------------------------------------------"
+		sleep 1
+
+		# Formatting is not needed as parted will create a formatted partition
+		echo "Formatting partitions"
+		mkfs -t vfat ${sd_dev}1
+		mkfs -t ext4 ${sd_dev}2
+		echo "-----------------------------------------------------------------------------"
+
+		echo "Mounting sd card"
+		mkdir -p /tmp/rz_sdm1
+		mkdir -p /tmp/rz_sdm2
+
+		mount ${sd_dev}1 /tmp/rz_sdm1
+		mount ${sd_dev}2 /tmp/rz_sdm2
+
+		sleep 2
+
+		if [[ -z $(df -hT | grep ${sd_dev}) ]]; then
+			echo "Device not mounted"
+			exit
+		else
+			echo "Device found"
+		fi
+
+		echo "-----------------------------------------------------------------------------"
+		echo "Copying files..."
+		echo "Changing to ${PWD}"
+		echo "Listing rootfs partition:"
+		ls -l
+		tar -xvjf ${rootfs_image_path} -C /tmp/rz_sdm2/
+		#cp rzpi.dtb /tmp/rz_sdm2/boot/
+		#ls -l /tmp/rz_sdm2/
+		cd /tmp/rz_sdm2/boot
+		#gzip -k Image-*
+		#mv Image-*.gz Image.gz
+		echo "Listing boot directory:"
+		ls -l
+		cp -rL ./* /tmp/rz_sdm1
```

Comment:
Please put a comment here as it will copies environment file and binary files to partition 1.